### PR TITLE
[GEOS-9462] Downgrade ArcSDE extension to a community module

### PR DESCRIPTION
**Pull request by groldan**
* Module refactored from src/extensions/arcsde to src/community/arcsde
* Module coordinates changed from `org.geoserver:gs-arcsde` to
`org.geoserver.community:gs-arcsde`
* Documentaion moved to community section, download instrucions
updated

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Committs changing the REST API, or any configuration object, should check it the REST API docs (Swagger YAML files and classic documentation) need to be updated.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by groldan** [comment]
Looks like it should all be in place, please review.

**Comment by groldan** [comment]
The geotools PR is still a draft: https://github.com/geotools/geotools/pull/2771

File: doc/en/user/source/community/arcsde/index.rst [hunk 1]
```diff
@@ -5,7 +5,7 @@ ArcSDE
...
 
-ESRI's `ArcSDE <http://www.esri.com/software/arcgis/arcsde/>`_ is a spatial engine that runs on top of a relational database such as Oracle or SQL Server.  GeoServer with the ArcSDE extension supports ArcSDE **versions 9.2 and 9.3**.  It has been tested with **Oracle 10g** and **Microsoft SQL Server 2000 Developer Edition**.  The ArcSDE extension is based on the GeoTools ArcSDE driver and uses the ESRI Java API libraries.  See the `GeoTools ArcSDE page <http://docs.geotools.org/latest/userguide/library/data/arcsde.html>`_ for more technical details.
+ESRI's `ArcSDE <http://www.esri.com/software/arcgis/arcsde/>`_ is a spatial engine that runs on top of a relational database such as Oracle or SQL Server.  GeoServer with the ArcSDE extension supports ArcSDE **versions 10.2 and 10.3**.  It has been tested with **Oracle 10g** and **Microsoft SQL Server 2000 Developer Edition**.  The ArcSDE extension is based on the GeoTools ArcSDE driver and uses the ESRI Java API libraries.  See the `GeoTools ArcSDE page <http://docs.geotools.org/latest/userguide/library/data/arcsde.html>`_ for more technical details.
```
Comment (aaime) [inline doc/en/user/source/community/arcsde/index.rst]: Will the documentation for arcsde move in geotools? Yup:
https://github.com/geotools/geotools/pull/2771/files#diff-6ec99f02e0b0eec76a4a64682f3181e6L90

Comment (groldan) [reply doc/en/user/source/community/arcsde/index.rst]: Actually that link doesn't change. Followed what's done with the other unsupported data plugins, they're listed under the unsupported title in the data page but are all located under library/data:
```

Unsupported plugins:

.. toctree::
   :maxdepth: 1
   
   arcsde
   dxf
   georest
   kml
   solr
   wfs-ng
```

See the links at the bottom of http://docs.geotools.org/latest/userguide/library/data/

**Pull request merged**